Description: Ensure that the local copy of expat_config.h is used, not the one installed by libexpat1-dev
Author: Olivier Tilloy <olivier.tilloy@canonical.com>

--- a/external/expat/UnpackedTarball_expat.mk
+++ b/external/expat/UnpackedTarball_expat.mk
@@ -15,6 +15,7 @@ $(eval $(call gb_UnpackedTarball_update_
 
 $(eval $(call gb_UnpackedTarball_add_patches,expat,\
 	external/expat/expat-winapi.patch \
+	external/expat/expat-include-config.patch \
 ))
 
 # This is a bit hackish.
--- /dev/null
+++ b/external/expat/expat-include-config.patch
@@ -0,0 +1,81 @@
+Description: Ensure that the local copy of expat_config.h is used, not the one installed by libexpat1-dev
+Author: Olivier Tilloy <olivier.tilloy@canonical.com>, Heather Ellsworth <heather.ellsworth@canonical.com>, Rico Tzschichholz <ricotz@ubuntu.com>
+Last-Updated: 2021-07-15
+
+--- a/expat-2.4.4/lib/xmlparse.c
++++ b/expat-2.4.4/lib/xmlparse.c
+@@ -58,7 +58,7 @@
+ 
+ #define XML_BUILDING_EXPAT 1
+ 
+-#include <expat_config.h>
++#include "../expat_config.h"
+ 
+ #if ! defined(_GNU_SOURCE)
+ #  define _GNU_SOURCE 1 /* syscall prototype */
+--- a/expat-2.4.4/lib/xmlrole.c
++++ b/expat-2.4.4/lib/xmlrole.c
+@@ -38,7 +38,7 @@
+    USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+ 
+-#include <expat_config.h>
++#include "../expat_config.h"
+ 
+ #include <stddef.h>
+ 
+--- a/expat-2.4.4/lib/xmltok.c
++++ b/expat-2.4.4/lib/xmltok.c
+@@ -43,7 +43,7 @@
+    USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+ 
+-#include <expat_config.h>
++#include "../expat_config.h"
+ 
+ #include <stddef.h>
+ #include <string.h> /* memcpy */
+--- a/expat-2.4.4/tests/chardata.c
++++ b/expat-2.4.4/tests/chardata.c
+@@ -34,7 +34,7 @@
+    USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+ 
+-#include <expat_config.h>
++#include "../expat_config.h"
+ #include "minicheck.h"
+ 
+ #include <assert.h>
+--- a/expat-2.4.4/tests/runtests.c
++++ b/expat-2.4.4/tests/runtests.c
+@@ -40,7 +40,7 @@
+    USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+ 
+-#include <expat_config.h>
++#include "../expat_config.h"
+ 
+ #if defined(NDEBUG)
+ #  undef NDEBUG /* because test suite relies on assert(...) at the moment */
+--- a/expat-2.4.4/tests/structdata.c
++++ b/expat-2.4.4/tests/structdata.c
+@@ -30,7 +30,7 @@
+    USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+ 
+-#include "expat_config.h"
++#include "../expat_config.h"
+ 
+ #include <assert.h>
+ #include <stdlib.h>
+--- a/expat-2.4.4/xmlwf/xmlfile.c
++++ b/expat-2.4.4/xmlwf/xmlfile.c
+@@ -37,7 +37,7 @@
+    USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+ 
+-#include <expat_config.h>
++#include "../expat_config.h"
+ 
+ #include <stdio.h>
+ #include <stdlib.h>
